include/sta/Search.hh

```diff
@@ -606,6 +607,7 @@ protected:
   ExceptionFrom *filter_from_;
   ExceptionTo *filter_to_;
   VertexSet *filtered_arrivals_;
+  UniqueLock filtered_arrivals_lock_;
```

Comment:
This should be std::mutex
See any of the other lock variables as an example.
UniqueLock is for locking the mutex.

Reply:
I presume then the declaration should be then `std::mutex filtered_arrivals_lock_;` in `Search.hh`, and the lock have the form of `std::lock_guard<sta::UniqueLock> guard(this->filtered_arrivals_lock_);`.

Unless I have assumed wrong then, this will produce a type error on `lock_guard` - the type of `lock_guard<T_>` should be the same as of the lock that is being guarded. I can change the implementation to `UniqueLock lock(this->filtered_arrivals_lock_);` but I understood from the first message that `lock_guard` is more desirable.

Reply:
There are 6 mutex locks defined in Search.hh.
Make this lock definition and use in Search.cc the same as all of the others.